src/frontend/src/scheduler/distributed/stage.rs

```diff
@@ -622,11 +632,16 @@ impl StageRunner {
     }
 
     async fn schedule_tasks_for_all(&mut self, shutdown_rx: ShutdownToken) -> SchedulerResult<()> {
+        let captured_context = CapturedContext {
+            // TODO(kexiang): We go through the plan to make sure only the neccessay context is captured.
```

Comment:
This sounds error-prone because it needs the programmer to *manually* ensure a context variable is provided if a function is used. I prefer to capture all variables no matter being used or not. 

After serialization, the context should be small. If we do want to save some bytes, we may omit the default values.

Reply:
Agree.